internal/storage/gcs/request.go

```diff
@@ -76,6 +76,22 @@ type CreateObjectRequest struct {
 	MetaGenerationPrecondition *int64
 }
 
+// A request to create an object, accepted by Bucket.CreateObject.
+type CreateObjectRequest struct {
+	CreateChunkUploaderRequest
+
+	// A reader from which to obtain the contents of the object. Must be non-nil.
+	Contents io.Reader
```

Comment:
Content is duplicated in CreateObjectRequest and CreateChunkUploaderRequest

Reply:
Good catch! Removed.